projects/emissary-ingress/emissary/helm/patches/0001-helm-changes.patch

```diff
@@ -4501,15 +4717,327 @@ index 2d06c8127..f1e043136 100644
+   labels:
+     app.kubernetes.io/name: {{ include "ambassador.name" . }}
+     {{- include "ambassador.labels" . | nindent 4 }}
+diff --git a/charts/emissary-ingress/templates/listener.yaml b/charts/emissary-ingress/templates/listener.yaml
+index 033051cc8..0a8d60e13 100644
+--- a/charts/emissary-ingress/templates/listener.yaml
++++ b/charts/emissary-ingress/templates/listener.yaml
+@@ -4,7 +4,7 @@ apiVersion: getambassador.io/v3alpha1
+ kind: Listener
+ metadata:
+   name: {{ include "ambassador.fullname" . }}-https-listener
+-  namespace: {{ include "ambassador.namespace" . }}
++  namespace: {{ .Release.Namespace | default .Values.defaultNamespace | quote }}
+ spec:
+   port: 8443
+   protocol: HTTPS
+@@ -14,14 +14,14 @@ spec:
+       from: ALL
+   {{- if hasKey .Values.env "AMBASSADOR_ID" }}
+   ambassador_id:
+-  - {{ .Values.env.AMBASSADOR_ID | quote }}
++  - {{ .Values.env.AMBASSADOR_ID |.Release.Namespace | default .Values.defaultNamespace | quote }}
+   {{- end }}
+ ---
+ apiVersion: getambassador.io/v3alpha1
+ kind: Listener
+ metadata:
+   name: {{ include "ambassador.fullname" . }}-http-listener
+-  namespace: {{ include "ambassador.namespace" . }}
++  namespace: {{ .Release.Namespace | default .Values.defaultNamespace | quote }}
+ spec:
+   port: 8080
+   protocol: HTTP
+@@ -31,6 +31,6 @@ spec:
+       from: ALL
+   {{- if hasKey .Values.env "AMBASSADOR_ID" }}
+   ambassador_id:
+-  - {{ .Values.env.AMBASSADOR_ID | quote }}
++  - {{ .Values.env.AMBASSADOR_ID |.Release.Namespace | default .Values.defaultNamespace | quote }}
+   {{- end }}
+ {{ end }}
+diff --git a/charts/emissary-ingress/templates/module.yaml b/charts/emissary-ingress/templates/module.yaml
+index a13d45996..e9287ac4a 100644
+--- a/charts/emissary-ingress/templates/module.yaml
++++ b/charts/emissary-ingress/templates/module.yaml
+@@ -3,7 +3,7 @@ apiVersion: getambassador.io/v2
+ kind: Module
+ metadata:
+   name: ambassador
+-  namespace: {{ include "ambassador.namespace" . }}
++  namespace: {{ .Release.Namespace | default .Values.defaultNamespace | quote }}
+   labels:
+     app.kubernetes.io/name: {{ include "ambassador.name" . }}
+     {{- include "ambassador.labels" . | nindent 4 }}
+@@ -12,7 +12,7 @@ metadata:
+ spec:
+   {{- if .Values.env }}
+   {{- if hasKey .Values.env "AMBASSADOR_ID" }}
+-  ambassador_id: {{ .Values.env.AMBASSADOR_ID | quote }}
++  ambassador_id: {{ .Values.env.AMBASSADOR_ID |.Release.Namespace | default .Values.defaultNamespace | quote }}
+   {{- end }}
+   {{- end }}
+   config:
+diff --git a/charts/emissary-ingress/templates/namespace.yaml b/charts/emissary-ingress/templates/namespace.yaml
+index 4535c74f2..c3cde61b4 100644
+--- a/charts/emissary-ingress/templates/namespace.yaml
++++ b/charts/emissary-ingress/templates/namespace.yaml
+@@ -4,5 +4,5 @@ kind: Namespace
+ metadata:
+   labels:
+     product: aes
+-  name: {{ include "ambassador.namespace" . }}
++  name: {{ .Release.Namespace | default .Values.defaultNamespace | quote }}
```

Comment:
We haven't talked about whether or not we should create namespaces automatically. Metallb doesn't do it, not sure about harbor. We should make it consistent.

Reply:
Yea that would be a good conversation